ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetadataReceivedListenerImpl.java

```diff
@@ -0,0 +1,254 @@
+package com.spectralogic.ds3client.metadata;
+
+import com.google.common.base.Joiner;
+import com.spectralogic.ds3client.helpers.MetadataReceivedListener;
+import com.spectralogic.ds3client.networking.Metadata;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.*;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MetadataReceivedListenerImpl implements MetadataReceivedListener {
+
+    private String localFilePath = null;
+
+    public MetadataReceivedListenerImpl(final String localFilePath) {
+        this.localFilePath = localFilePath;
+    }
+
+    @Override
+    public void metadataReceived(final String filename, final Metadata metadata) {
+        restoreMetaData(filename, metadata);
+    }
+
+    /**
+     * Restore the metadata to local file
+     *
+     * @param objectName name of the file to be restored
+     * @param metadata   metadata which needs to be set on local file
+     */
+    private void restoreMetaData(final String objectName, final Metadata metadata) {
+        final StringBuilder builder = new StringBuilder();
+        final Joiner joiner = Joiner.on(", ");
+
+        builder.append("Metadata for object ").append(objectName).append(": ");
+        final String os = System.getProperty("os.name");
+
+        String creationTime = null;
+        String accessTime = null;
+        String modifiedTime = null;
+        String owner = null;
+        String groupName = null;
+        String storedOS = null;
+
+        if (metadata.get("ds3-creationTime").size() > 0)
```

Comment:
Please add explicit curly braces to conform with our java coding standard.  Here is a link to the specific section on the topic: [Curly Braces](https://github.com/SpectraLogic/spectralogic.github.com/wiki/Java-Style-Guide#conditional-statements)